src/security/forward.h

```diff
@@ -101,9 +101,24 @@ typedef std::list<Security::CertPointer> CertList;
 typedef std::list<Security::CrlPointer> CertRevokeList;
 
 #if USE_OPENSSL
+CtoCpp1(EVP_PKEY_free, EVP_PKEY *)
+typedef Security::LockingPointer<EVP_PKEY, EVP_PKEY_free_cpp, HardFun<int, EVP_PKEY *, EVP_PKEY_up_ref> > PrivateKeyPointer;
+#elif USE_GNUTLS
+typedef std::shared_ptr<struct gnutls_x509_privkey_int> PrivateKeyPointer;
```

Comment:
nit: is there any reason for using typedef instead of using?

Reply:
@rouskov what do you mean by thumbs down on my answer? its true.

Reply:
> what do you mean by thumbs down on my answer? its true.

While the statement in your answer may be correct, the answer does not address the problem implied by the change request. In this specific case, this is a very minor potential problem in a long-suffering PR, so I did not do more about it, but the overall flaw is very common -- folks are focusing on defending their actions/code/words instead of focusing on improving the PR. In my estimates, our PR process efficiency is lower than 30% -- most of the time spent on PRs is wasted. This amount of waste is ridiculous IMO.

I hope my [EasierDoneThanSad](https://squid.zulipchat.com/#narrow/stream/314059-core/topic/Easier.20Done.20.28Than.20Said.29/near/286106470) experiment with Francesco will improve things, but it cannot solve many of the problems, and I suspect you are [against](https://github.com/squid-cache/squid/pull/864#issuecomment-1100425996) that kind of collaboration anyway.

Reply:
Done in [d4a42e7](https://github.com/squid-cache/squid/pull/694/commits/d4a42e7ecfe4f517704031f6e7827ccb21512d08)